pkg/controller/stack/stack_controller.go

```diff
@@ -781,48 +785,91 @@ func (sess *reconcileStackSession) DestroyStack() error {
-		return nil, err
-	}
+		if sess.stack.GitAuth.SSHAuth != nil {
+			privateKey, err := sess.resolveResourceRef(&sess.stack.GitAuth.SSHAuth.SSHPrivateKey)
+			if err != nil {
+				return nil, errors.Wrap(err, "resolving gitAuth SSH private key")
+			}
+			gitAuth.SSHPrivateKey = privateKey
+
+			password, err := sess.resolveResourceRef(sess.stack.GitAuth.SSHAuth.Password)
+			if err != nil {
+				return nil, errors.Wrap(err, "resolving gitAuth SSH password")
+			}
+			gitAuth.Password = password
 
-	var gitAuth *auto.GitAuth
+			return gitAuth, nil
+		}
 
-	// First check if an SSH private key has been specified.
-	if sshPrivateKey, exists := secret.Data["sshPrivateKey"]; exists {
-		gitAuth = &auto.GitAuth{
-			SSHPrivateKey: string(sshPrivateKey),
+		if sess.stack.GitAuth.BasicAuth == nil {
+			return nil, errors.New("gitAuth config must specify exactly one of " +
+				"'personalAccessToken', 'sshPrivateKey' or 'basicAuth'")
 		}
 
-		if password, exists := secret.Data["password"]; exists {
-			gitAuth.Password = string(password)
+		userName, err := sess.resolveResourceRef(&sess.stack.GitAuth.BasicAuth.UserName)
+		if err != nil {
+			return nil, errors.Wrap(err, "resolving gitAuth username")
 		}
 
-		// Add the project repo's public SSH keys to the SSH known hosts
-		// to perform the necessary key checking during SSH git cloning.
-		sess.addSSHKeysToKnownHosts(sess.stack.ProjectRepo)
-		// Then check if a personal access token has been specified.
-	} else if accessToken, exists := secret.Data["accessToken"]; exists {
-		gitAuth = &auto.GitAuth{
-			PersonalAccessToken: string(accessToken),
+		password, err := sess.resolveResourceRef(&sess.stack.GitAuth.BasicAuth.Password)
+		if err != nil {
+			return nil, errors.Wrap(err, "resolving gitAuth password")
+		}
+
+		gitAuth.Username = userName
+		gitAuth.Password = password
+	} else if sess.stack.GitAuthSecret != "" {
+		namespacedName := types.NamespacedName{Name: sess.stack.GitAuthSecret, Namespace: sess.namespace}
+
+		// Fetch the named secret.
+		secret := &corev1.Secret{}
+		if err := sess.kubeClient.Get(context.TODO(), namespacedName, secret); err != nil {
+			sess.logger.Error(err, "Could not find secret for access to the git repositoriy",
+				"Namespace", sess.namespace, "Stack.GitAuthSecret", sess.stack.GitAuthSecret)
+			return nil, err
 		}
-		// Then check if basic authentication has been specified.
-	} else if username, exists := secret.Data["username"]; exists {
-		if password, exists := secret.Data["password"]; exists {
+
+		// First check if an SSH private key has been specified.
+		if sshPrivateKey, exists := secret.Data["sshPrivateKey"]; exists {
 			gitAuth = &auto.GitAuth{
-				Username: string(username),
-				Password: string(password),
+				SSHPrivateKey: string(sshPrivateKey),
+			}
+
+			if password, exists := secret.Data["password"]; exists {
+				gitAuth.Password = string(password)
+			}
+
+			// Then check if a personal access token has been specified.
```

Comment:
Looks like you might need to run `gofmt` on this.

Reply:
Seems like this is an artifact of being the if else block (I did run go fmt).